modules/rubiconBidAdapter.js

```diff
@@ -1305,4 +1186,70 @@ export function resetUserSync() {
   hasSynced = false;
 }
 
+function setBidFloors(bidRequest, imp) {
+  if (imp.bidfloorcur != 'USD') {
+    delete imp.bidfloor;
+    delete imp.bidfloorcur;
+  }
+
+  let bidFloor = parseFloat(deepAccess(bidRequest, 'params.floor'));
+
+  if (!isNaN(bidFloor)) {
+    imp.bidfloor = bidFloor;
+    imp.bidfloorcur = 'USD';
+  }
+}
+
+function addMultibid(data, multibid) {
+  if (!multibid) {
+    return;
+  }
+  deepSetValue(data, 'ext.prebid.multibid', multibid.reduce((result, i) => {
+    let obj = {};
+
+    Object.keys(i).forEach(key => {
+      obj[key.toLowerCase()] = i[key];
+    });
+
+    result.push(obj);
+
+    return result;
+  }, []));
+}
+
+function addOrtbFirstPartyData(data, nonBannerRequests) {
+  let fpd = {};
+  const keywords = new Set();
+  nonBannerRequests.forEach(bidRequest => {
+    const bidFirstPartyData = {
+      user: {ext: {data: {...bidRequest.params.visitor}}},
+      site: {ext: {data: {...bidRequest.params.inventory}}}
+    };
+
+    // add site.content.language
+    const impThatHasVideoLanguage = data.imp.find(imp => imp.ext?.prebid?.bidder?.rubicon?.video?.language);
+    if (impThatHasVideoLanguage) {
+      bidFirstPartyData.site.content = {
+        language: impThatHasVideoLanguage.ext?.prebid?.bidder?.rubicon?.video?.language
+      }
+    }
+
+    if (bidRequest.params.keywords) {
+      const keywordsArray = (!Array.isArray(bidRequest.params.keywords) ? bidRequest.params.keywords.split(',') : bidRequest.params.keywords);
+      keywordsArray.forEach(keyword => keywords.add(keyword));
+    }
+    fpd = mergeDeep(fpd, bidRequest.ortb2 || {}, bidFirstPartyData);
+
+    // add user.id from config.
+    // NOTE: This is DEPRECATED. user.id should come from setConfig({ortb2}).
+    const configUserId = config.getConfig('user.id');
+    fpd.user.id = fpd.user.id || configUserId;
+  });
+
+  mergeDeep(data, fpd);
+
+  deepSetValue(data, 'site.keywords', Array.from(keywords.values()).join(','));
```

Comment:
If `keywords` is empty this will pass a empty string to PBS. I think we do not want to send in if this is empty.

Reply:
wrapped around an "if keywords not empty"

Reply:
👍